File: isabelle.eclipse.launch/src/isabelle/eclipse/launch/tabs/SingleCheckedListener.java
package isabelle.eclipse.launch.tabs;

import org.eclipse.jface.viewers.CheckStateChangedEvent;
import org.eclipse.jface.viewers.CheckboxTableViewer;
import org.eclipse.jface.viewers.ICheckStateListener;

public class SingleCheckedListener implements ICheckStateListener {

	private final CheckboxTableViewer viewer;
	
	public SingleCheckedListener(CheckboxTableViewer viewer) {
		this.viewer = viewer;
	}

	@Override
	public void checkStateChanged(CheckStateChangedEvent event) {
		viewer.setCheckedElements(new Object[] { event.getElement() });
	}

}

File: isabelle.eclipse.launch/src/isabelle/eclipse/launch/tabs/SingleCheckedListener.scala
package isabelle.eclipse.launch.tabs

import org.eclipse.jface.viewers.{CheckStateChangedEvent, CheckboxTableViewer, ICheckStateListener}

/**
 * A listener for CheckboxTableViewer that ensures only a single element is selected.
 * 
 * Whenever a new element is selected, the other ones are deselected.
 * 
 * @author Andrius Velykis
 */
class SingleCheckedListener(viewer: CheckboxTableViewer)
    extends ICheckStateListener {

  override def checkStateChanged(event: CheckStateChangedEvent) {
    if (event.getChecked) {
      viewer.setCheckedElements(Array(event.getElement()))
    }
  }

}
